### DISCUSSION TITLE
AP2 Build request

### DISCUSSION THREAD

**Discussion by antonellocaroli** [mikebrady/shairport-sync#1235] [General]
Is there a particular reason why the nqptp source is required to be present in order to compile shairport-sync?

Why not include the files it needs to compile directly in the shairport-sync source?

**Comment by mikebrady** [comment]
Thanks for the post. `nqptp` is needed for Shairport Sync to function. It provides information via a SMI and the layout is defined in a header file in the `nqptp` repository. Shairport Sync needs that file. So, on the basis that you would normally be building both programs, this seems to be a sensible way to organise it. We might separate them fully later on. Is it a problem?

**Comment by mikebrady** [comment]
The dependency on the presence of the `nqptp` folder has been removed. Of course, `nqptp` must be installed and running.
